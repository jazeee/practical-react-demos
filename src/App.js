import React, { Component } from 'react';
import GithubUser from './demo/github-user.jsx';
import './App.css';
import {DemoButton} from './bootstrap4/demo-button.jsx';
import {DemoModal} from './bootstrap4/demo-modal.jsx';
import {Header} from './nav/header.jsx';

class App extends Component {
	constructor(props) {
		super(props);
		this.state = {
			isModalOpen: false,
		};
	}
	toggleModal = () => {
		const isModalOpen = !this.state.isModalOpen;
		this.setState({isModalOpen});
	}
		render = () => {
			return (
				<div className='App'>
					<Header />
					<GithubUser />
					<DemoButton
						onClick={this.toggleModal}
						text='Hello Jaz'
					/>
					<DemoModal
						isOpen={this.state.isModalOpen}
						toggleModal={this.toggleModal}
						title='Hello'
						body='Welcome to modals, Jaz.'
					/>
				</div>
			);
		}
}

export default App;
