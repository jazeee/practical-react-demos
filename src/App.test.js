import React from 'react';
import {App} from './App';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

it('renders without crashing', () => {
	shallow(<App />);
});

it('renders correctly', () => {
	const wrapper = shallow(
		<App/>
	);

	expect(toJson(wrapper)).toMatchSnapshot();
});
