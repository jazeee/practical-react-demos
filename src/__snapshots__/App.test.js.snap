// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <Header />
  <GithubUser />
  <Component
    onClick={[Function]}
    text="Hello Jaz"
  />
  <Component
    body="Welcome to modals, Jaz."
    isOpen={false}
    title="Hello"
    toggleModal={[Function]}
  />
</div>
`;
